transactor/src/main/scala/io/mediachain/datastore/PersistentDatastore.scala

```diff
@@ -0,0 +1,92 @@
+package io.mediachain.datastore
+
+import java.util.Random
+import java.util.concurrent.LinkedBlockingDeque
+import com.amazonaws.AmazonServiceException
+import org.slf4j.{Logger, LoggerFactory}
+import io.mediachain.multihash.MultiHash
+
+class PersistentDatastore(config: PersistentDatastore.Config)
+  extends BinaryDatastore with AutoCloseable with Runnable {
+  
+  val logger = LoggerFactory.getLogger(classOf[PersistentDatastore])
+  val dynamo = new DynamoDatastore(config.dynamo)
+  val rocks = new RocksDatastore(config.rocks)
+  val random = new Random
+  val maxBackoffRetry = 60 // second
+  val queue = new LinkedBlockingDeque[MultiHash]
+  val writer = new Thread(this, "PersistentDataStore#write")
```

Comment:
@vyzo I'm fairly sure using Futures requires a thread pool (by default the ForkJoinPool from the global ExecutionContext)

Given that the order of writes to the external datastore doesn't really matter as long as they all complete eventually, and that you're retrying/queueing anyway, I don't see why a small number of concurrent writers isn't appropriate.